### PR TITLE
Change workspace cleanup to async.

### PR DESCRIPTION
**Pull request by justinlittman**
refs https://github.com/sul-dlss/argo/issues/2752

## Why was this change made? 🤔
Avoid timeouts

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

File: app/services/cleanup_service.rb [hunk 1]
```diff
@@ -22,19 +22,11 @@ def self.cleanup_workspace_content(druid, base)
...
     bag_dir = File.join(Settings.cleanup.local_export_home, id)
-    remove_branch(bag_dir)
+    FileUtils.rm_rf(bag_dir)
     tarfile = "#{bag_dir}.tar"
-    remove_branch(tarfile)
+    FileUtils.rm_f(tarfile)
   end
   private_class_method :cleanup_export
-
-  # @param [Pathname,String] pathname The full path of the branch to be removed
```
Comment (justinlittman) [inline app/services/cleanup_service.rb:33]: With the original code, when invoked from a job, an error was being thrown on `pathname.rmtree` claiming that the file does not exist. Logging confirmed that `pathname.exist?` returns true.

My theory is that the Reset Workspace workflow step (prior to Clean Workspace) is deleting the directory. However, when Clean Workspace was changed to a job, it is now performed on a different server (the worker server) and there is an NFS sync latency. (Thus, this service thinks that the directory exists, but when it goes to delete the directory, it doesn't exist.)

This changes side-steps the problem by using a forgiving delete.

File: spec/jobs/cleanup_job_spec.rb [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+require 'rails_helper'
+
+RSpec.describe CleanupJob, type: :job do
+  subject(:perform) do
+    described_class.perform_now(druid:, background_job_result: result, workflow:)
+  end
+
+  let(:druid) { 'druid:mk420bs7601' }
+  let(:result) { create(:background_job_result) }
+  # let(:item) { instance_double(Cocina::Models::DRO) }
```
Comment (lwrubel) [inline spec/jobs/cleanup_job_spec.rb:12]: Cruft?

**Review by lwrubel — APPROVED** [review]
Approved, noting some possible cruft.

File: spec/jobs/cleanup_job_spec.rb (hunk 2 above)
Comment (justinlittman) [reply spec/jobs/cleanup_job_spec.rb:12]: Yup. Thanks.

**Pull request merged**